psi4/driver/driver_cbs.py

```diff
@@ -916,58 +1077,208 @@ def return_energy_components():
 
     # Integrate CFOUR methods
     VARH.update(cfour_psivar_list())
+
+    # Integrate density functionals
+    for funcname in functionals:
+        if funcname == "hf" or funcname == "scf":
+            continue
+        VARH[funcname] = {
+                         funcname: 'DFT TOTAL ENERGY',
+                f'{funcname}-dft': 'DFT TOTAL ENERGY',
+               f'{funcname}-fctl': 'DFT FUNCTIONAL TOTAL ENERGY'}
+        if 'dispersion' in functionals[funcname].keys():
+            if functionals[funcname]['dispersion']['type'] == 'nl':
+                VARH[funcname][f'{funcname}-nl'] = 'DFT VV10 ENERGY'
+            else:
+                VARH[funcname][f'{funcname}-disp'] = 'DISPERSION CORRECTION ENERGY'
+        if 'c_mp2' in functionals[funcname].keys():
+            VARH[funcname][f'{funcname}-dh'] = 'DOUBLE-HYBRID CORRECTION ENERGY'
     return VARH
     # yapf: enable
 
 
 VARH = return_energy_components()
 
 
-def _get_default_xtpl(nbasis: int, xtpl_type: str) -> Callable:
+def _get_default_xtpl(nbasis: int, xtpl_type: Union[str, None]) -> Callable:
     """ A helper function to determine default extrapolation type.
 
     Parameters
     ----------
     nbasis
         Number of basis sets
-    xtpl_type
-        {'scf', 'corl'}
-        Extrapolation type: 'scf' for the total energy, 'corl' for just the
-        correlation component.
+    xtpl_type : {'scf', 'corl', 'fctl', 'dh', None}
```

Comment:
```suggestion
    xtpl_type
        {'scf', 'corl', 'fctl', 'dh', None}
```
sphinx wants choices on newline

Reply:
That was `yapf`...

Reply:
ok, weird that a formatter is messing with docstrings.